src/main/webapp/app/iris/http-message.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { IrisClientMessage, IrisMessage } from 'app/entities/iris/iris.model';
+
+type EntityResponseType = HttpResponse<IrisMessage>;
+type EntityArrayResponseType = HttpResponse<IrisMessage[]>;
+
+/**
+ * Provides a singleton instance of http client to perform CRUD operations.
+ */
+@Injectable({ providedIn: 'root' })
+export class IrisHttpMessageService {
+    public resourceUrl = 'api/iris/sessions';
+
+    constructor(private httpClient: HttpClient) {}
+
+    /**
+     * creates a message for a session
+     * @param {number} sessionId
+     * @param {IrisClientMessage} message
+     * @return {Observable<EntityResponseType>}
+     */
+    createMessage(sessionId: number, message: IrisMessage): Observable<EntityResponseType> {
+        return this.httpClient.post<IrisClientMessage>(`${this.resourceUrl}/${sessionId}/messages`, message, { observe: 'response' });
```

Comment:
the sender of the response is either ARTEMIS or LLM, so I think let's just return IrisMessage here to be general
```suggestion
        return this.httpClient.post<IrisMessage>(`${this.resourceUrl}/${sessionId}/messages`, message, { observe: 'response' });
```

Reply:
I adjusted IrisServerMessage, now is sender: IrisSender.SERVER | IrisSender.SYSTEM;

Reply:
then the response should be from server right?  `return this.httpClient.post<IrisServerMessage>`

Reply:
No, IrisServerMessage is for messages with sender artemis or llm. Create message is for student messages.

Reply:
ah right 👍  I forgot that the server won't return LLM's messages via REST